### PR TITLE
Removes all mysql_* calls outside Elgg\Database, and removes old upgrade scripts

### LINKED ISSUE #5703: Put all mysql_* calls behind the API
This is step towards DBAL.

There several issues:
- [ ] upgrade scripts
- [ ] tests
- [ ] installer
- [ ] Elgg_CacheHandler
- [ ] mod/profile/icondirect.php

Id like all of them to use Elgg_Database methods instead of mysql_* directly, so we could abstract DB layer inside this class only.

### PR DESCRIPTION
**Pull request by mrclay**
Fixes #5703

### CODE REVIEW

**Comment by ewinslow** [comment]
THIS WOULD BE AWESOME TO HAVE DONE FOR THE HACKATHON PLEASE FINISH IT

**Comment by ewinslow** [comment]
:)

**Comment by mrclay** [comment]
I split this into 2 commits. Have not yet testing the installer

File: engine/classes/Elgg/CacheHandler.php [hunk 1]
```diff
@@ -138,28 +138,32 @@ protected function setupSimplecache() {
 			return;
 		}
```
Comment (mrclay) [inline engine/classes/Elgg/CacheHandler.php]: This could be done in 1.11

File: engine/classes/ElggInstaller.php [hunk 2]
```diff
@@ -1212,30 +1212,36 @@ protected function validateDatabaseVars($submissionVars, $formVars) {
 	 * @return bool
 	 */
 	protected function checkDatabaseSettings($user, $password, $dbname, $host) {
```
Comment (mrclay) [inline engine/classes/ElggInstaller.php]: Also could go in 1.11

**Pull request closed without merging**